uui-core/src/hooks/useLayer.ts

```diff
@@ -0,0 +1,13 @@
+import { useEffect, useRef } from 'react';
+import { useUuiContext } from '../services';
+
+export const useLayer = () => {
+    const context = useUuiContext();
+    const layer = useRef(context.uuiLayout?.getLayer()).current;
+
+    useEffect(() => {
+        return () => layer && context.uuiLayout?.releaseLayer(layer);
+    }, [context]);
+
+    return layer?.zIndex;
```

Comment:
let's return the whole layer object, it will be more versatile and better match with hook name.